### PR TITLE
feat: add vaadin-spreadsheet-flow component

### PR DESCRIPTION
**Pull request by manolo**
It should merged instead of squash because brings git history from the original addon repository.

Fixes https://github.com/vaadin/spreadsheet/issues/824 
Closes https://github.com/vaadin/platform/issues/2027

### CODE REVIEW

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+  //
+  function maybeStartModule() {
+    // TODO: it may not be necessary to check gwtOnLoad here.
+    if (gwtOnLoad && bodyDone) {
+      gwtOnLoad(onLoadErrorFunc, '__MODULE_NAME__', base, softPermutationId);
+    }
+  }
+
+  // Customized for Flow, replaced with the computeScriptBase() function
+  __COMPUTE_SCRIPT_BASE__
```
[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

**Comment by sissbruecker** [comment]
Can we do something about the SonarCloud issues:

Bildschirmfoto 2022-04-28 um 11 34 43

Currently we at least have 0 bugs and 0 security issues, so these are some metric that are somewhat working in this repo. With adding this change the SonarCloud metrics basically become useless, and every PR would fail validation. Solving this  doesn't necessarily mean we have to address every individual issue right now. Maybe we can exclude the module from SonarCloud checks for now, and add a ticket to resolve this at a later stage.

**Comment by manolo** [comment]
@sissbruecker I already fixed critical sonar errors, let's see if it passes now

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

File: scripts/build.sh [hunk 2]
```diff
@@ -39,7 +41,11 @@ then
   modified=`curl -s https://api.github.com/repos/vaadin/flow-components/pulls/$PR/files \
     | jq -r '.[] | .filename' | grep 'vaadin.*parent' | perl -pe 's,^vaadin-(.*)-flow-parent.*,$1,g' | sort -u`
 
-  if [ `echo "$modified" | wc -w` -lt 5 ] && [ `echo ${#modifiedAll}` = `echo ${#modifiedComponent}` ]
+  nmods=`echo "$modified" | wc -w`
+  [ $nmods -eq 1 ] && echo "$title" | grep -q " vaadin-$modified-flow " && partial=yes
+  [ -z partial ] && [ $nmods -lt 5 ] && [ `echo ${#modifiedAll}` = `echo ${#modifiedComponent}` ] && partial=yes
```
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh:46]: @sonatype-lift ignore

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

File: scripts/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh]: @sonatype-lift ignore

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift

[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

File: scripts/build.sh [hunk 3]
```diff
@@ -39,7 +41,11 @@ then
   modified=`curl -s https://api.github.com/repos/vaadin/flow-components/pulls/$PR/files \
```
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh]: @sonatype-lift ignore

File: scripts/build.sh [hunk 4]
```diff
@@ -39,7 +41,11 @@ then
...
     | jq -r '.[] | .filename' | grep 'vaadin.*parent' | perl -pe 's,^vaadin-(.*)-flow-parent.*,$1,g' | sort -u`
 
-  if [ `echo "$modified" | wc -w` -lt 5 ] && [ `echo ${#modifiedAll}` = `echo ${#modifiedComponent}` ]
+  nmods=`echo "$modified" | wc -w`
+  [ $nmods -eq 1 ] && echo "$title" | grep -q " vaadin-$modified-flow " && partial=yes
```
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh]: @sonatype-lift ignore

File: scripts/build.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh]: @sonatype-lift ignore

File: scripts/build.sh [hunk 5]
```diff
@@ -39,7 +41,11 @@ then
```
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh:40]: @sonatype-lift ignore

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (manolo) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/resources/com/vaadin/component/spreadsheet/client/SpreadsheetSingleScriptTemplate.js:105]: @sonatype-lift ignore

File: scripts/build.sh [hunk 6]
```diff
@@ -19,31 +19,37 @@ then
```
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/build.sh]: @sonatype-lift ignore

**Review by yuriy-fix — APPROVED** [review]

**Pull request merged**